### PR TITLE
Update usages of the source file name when copying a banner to a new file.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/32792

### CODE REVIEW

File: src/Features/Core/Portable/AddFileBanner/AbstractAddFileBannerCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -66,19 +71,47 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
...
+        /// <summary>
+        /// Looks at <paramref name="banner"/> to see if it contains the name of <paramref name="sourceDocument"/>
+        /// in it.  If so, those names will be replaced with <paramref name="destinationDocument"/>'s name.
+        /// </summary>
+        private ImmutableArray<SyntaxTrivia> UpdateEmbeddedFileNames(
+            Document sourceDocument, Document destinationDocument, ImmutableArray<SyntaxTrivia> banner)
+        {
+            var sourceName = IOUtilities.PerformIO(() => Path.GetFileName(sourceDocument.FilePath));
+            var destinationName = IOUtilities.PerformIO(() => Path.GetFileName(destinationDocument.FilePath));
+            if (string.IsNullOrEmpty(sourceName) || string.IsNullOrEmpty(destinationName))
```
Comment (genlu) [inline src/Features/Core/Portable/AddFileBanner/AbstractAddFileBannerCodeRefactoringProvider.cs]: nit: inline `PerformIO` so we would bail early if first call failed.

File: src/EditorFeatures/CSharpTest/AddFileBanner/AddFileBannerTests.cs [hunk 2]
```diff
@@ -190,6 +190,61 @@ static void Main()
...
 
+class Program2
+{
+}
+        </Document>
+    </Project>
+</Workspace>");
+        }
+
+        [WorkItem(32792, "https://github.com/dotnet/roslyn/issues/32792")]
```
Comment (genlu) [inline src/EditorFeatures/CSharpTest/AddFileBanner/AddFileBannerTests.cs]: Could you add a test with multi-line comment banner please

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/AddFileBanner/AddFileBannerTests.cs]: done.

File: src/Features/Core/Portable/AddFileBanner/AbstractAddFileBannerCodeRefactoringProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/AddFileBanner/AbstractAddFileBannerCodeRefactoringProvider.cs]: enh :D

**Comment by CyrusNajmabadi** [comment]
@jinujoseph @mavasani this is good to merge in.

**Pull request merged**